src/adaptors.rs

```diff
@@ -176,6 +176,13 @@ impl<I> PutBack<I> where
         PutBack{top: None, iter: it}
     }
 
+    /// Create a `PutBack` along with the `value` to put back.
+    #[inline]
+    pub fn value(value: I::Item, it: I) -> Self
```

Comment:
Can we change this to for example `with_value`? It's more clear it's a constructor then.